lldb/test/API/functionalities/data-formatter/data-formatter-stl/msvcstl/string/TestDataFormatterMsvcStlString.py

```diff
@@ -0,0 +1,120 @@
+# coding=utf8
+"""
+Test std::*string summaries with MSVC's STL.
+"""
```

Comment:
I might've missed some discussion on this:

Not a blocker for this PR but why do we have separate API test directories for the various STLs? The only difference in them being `USE_LIBCPP` vs. `USE_LIBSTCPP` in the Makefiles. And now a third one with no apparent difference in the Makefile.

We should probably consolidate them all into the `data-formatter/data-formatter-stl` and just pass the necessary makefile flags from within the python tests (via `self.build`)?

@labath @DavidSpickett anything stopping us from doing that?

Reply:
I think it's mainly historical. We have `data-formatter/data-formatter-stl/generic/`, but it's a fairly recent addition. I think it would be nice to move everything towards that, in particular because it encourages a consistent presentation of different implementations -- not just for humans, but this also makes it easier to write data formatters for other data structures (which contain std:: types internally) -- it's nice when the data formatter works regardless of the STL version your code is compiled with.

In theory there could be tests that only make sense on a particular (e.g. a test for what happens when a particular internal pointer is corrupted), but those could be handled with ifdefs, or with separate tests.